perfkitbenchmarker/benchmarks/fio_benchmark.py

```diff
@@ -398,7 +400,7 @@ def Run(benchmark_spec):
 
   vm.PushFile(job_file_path, REMOTE_JOB_FILE_PATH)
 
-  if FLAGS.against_device:
+  if FLAGS.fio_target_mode in AGAINST_DEVICE_MODES:
```

Comment:
Done. I also added a parallel function for checking if we fill the target device, called `FillTarget`.